### DISCUSSION TITLE
Change case in output

### DISCUSSION THREAD

**Discussion by rhb327** [houmain/keymapper#77] [Q&A] [answered]
Trying to use for a simple abbreviation. I'm unable able to get multiple cases in the output using Shift{X} or CapsLock X CapsLock Y. The below will not work as expected and often end up as:

thanks,
rich

instead of

Thanks,
-Rich

```
# Signatures
Shift{T} H      >> Shift{T} H A N K S Comma Enter Minus Shift{R} I C H
#Shift{T} H      >> CapsLock T CapsLock H A N K S Comma Enter Minus CapsLock R CapsLock I C H
```
Any thoughts?

Thanks,
-Rich

**Comment by houmain** [comment]
Hi Rich,
Sorry, I could not reproduce this, neither on Linux nor Windows.
Have you tested it with this mapping alone or only with others?

Now I am thinking about supporting strings as output sequences, like:
`Shift{T} H >> "Thanks,\n-Rich"`

Cheers!

**Comment by rhb327** [comment]
Most things are working.  I have a pretty easy setup as follows:

[system="Linux"]
Shift   >> Shift
Control >> Control
AltLeft >> AltLeft

# Date shortcuts
D T T    >> $(date +%m/%d/%Y -d "+1 days") ^
D T Y    >> $(date +%m/%d/%Y -d "-1 days") ^
D T W    >> $(date +%m/%d/%Y -d "+7 days") ^
D T      >> $(date +%m/%d/%Y) ^

# Signatures
Shift{T} H      >> Shift{T} H A N K S Comma Enter Minus Shift{R} I C H
#Shift{T} H      >> CapsLock T CapsLock H A N K S Comma Enter Minus
CapsLock R CapsLock I C H

But for signatures the above with output two different variants depending
on what the caps lock state is:
thanks,
-rich

or

THANKS,
-RICH

I'm unable (so far) to achieve:

Thanks,
-Rich

**Comment by houmain** [comment]
Interesting, aren't these mappings interfering with your normal typing?
Can you see or insert the output of `$(date +%m/%d/%Y -d "+1 days")` somehow?

**Comment by rhb327** [comment]
Yes, all the date ones seem to work just fine!  I've used these for a long
time and generally the combinations are safe for me!

The "Th" can be an issue on occasion.

I used to use these in AutoKey but recently started to use Wayland.

Rich

**Comment by houmain** [comment]
I am curious, where can you see the output of the date command? In some terminal window?
Never implemented that feature. :)

**Comment by rhb327** [comment]
See attached.  Left shows "Th" and "tH" in gedit but right shows where the
dates go (to the terminal) that spawned keymmaper even when I type "dt" in
gedit.

It sounds like I'm in a unintended use category here so maybe not worth
pursuing!

-Rich

**Reply by ristomatti** [reply]
@rhb327 In case you're still in need for something like the date shortcuts, I think [Espanso](https://espanso.org/) would be the right tool for the job. I use both Keymapper and Espanso. I haven't yet been able to imagine too many uses for Espanso, but I use it specifically for adding timestamps on backup file names. I'm also using it to write Jira issues using Markdown syntax instead of it's native syntax I always forget. 🙂

**Reply by houmain** [reply, marked as answer]
With `keymapperctl --type` in 4.4.0 one could insert dates like:
```
D T T >> $(keymapperctl --type "$(date +%m/%d/%Y -d '+1 days')") ^
```
and concerning the mixed up case, you may want to retry it with the string typing feature:
```
Shift{T} H >> "Thanks, Rich"
```

**Reply by ristomatti** [reply]
@houmain The release notes looked awesome. I've yet to test it but will soon. :slightly_smiling_face:

**Reply by ristomatti** [reply]
Works great!
